core/src/main/java/org/mule/runtime/core/internal/config/togglz/scope/MuleTogglzFeatureScope.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.config.togglz.scope;
+
+import org.mule.runtime.core.internal.config.togglz.scope.type.MuleTogglzFeatureScopeType;
+import org.mule.runtime.core.internal.config.togglz.state.MuleTogglzFeatureStateRepository;
+
+/**
+ * The scope for the state of a {@link org.togglz.core.Feature}. In each scope, the
+ * {@link org.togglz.core.repository.FeatureState} can vary.
+ *
+ * @see MuleTogglzFeatureStateRepository
+ */
+public interface MuleTogglzFeatureScope {
```

Comment:
what is the purpose of this inte3rface? why nopt use the `MuleTogglzFeatureScopeType ` directly?

Reply:
This indicate the scope. The type indicate the type of scope.
For example. MuleTogglzArtifactFeatureScope has type ARTIFACT_SCOPE_TYPE and would have the artifactId
"http-test-app". 
Adding other types of scopes for example LocationTogglzArtifactFeatureScope would have another type and another way to define the granularity (for example ttp-test-app/source/processors/0 or http-test-app/source/*)